### PR TITLE
remove metadata on entity death, and when items are removed from pede…

### PR DESCRIPTION
**Pull request by PhanaticD**
…stal. Also remove references to "item_placed" I do not see anywhere where this key is actually set, maybe leftover from old code?

I am currently testing this on my server, can wait a few days if you do not trust it for me to notice any harm

the metadata removal is necessary otherwise it will be a memory leak

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/AncientAltarTask.java [hunk 1]
```diff
@@ -129,8 +129,7 @@ private void checkPedestal(Block pedestal) {
 
             itemLock.remove(item);
             item.remove();
-
-            pedestal.removeMetadata("item_placed", SlimefunPlugin.instance);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/AncientAltarTask.java:133]: Why was this removed?

Comment (PhanaticD) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/AncientAltarTask.java:133]: I could not find anywhere that this was actually added

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/AncientAltarTask.java:133]: Hmm true. Weird...
Since there hasn't been any bug reports or complaints about the altars recently I will just assume that this is safe to remove, some probably added this at some point and removed it half-heartedly, leaving these things behind. Guess it wasn't that important then.
